extmod/modjson.c

```diff
@@ -157,6 +157,7 @@ STATIC mp_obj_t mod_json_load(mp_obj_t stream_obj) {
     const mp_obj_type_t *stack_top_type = NULL;
     mp_obj_t stack_key = MP_OBJ_NULL;
     S_NEXT(s);
+    unsigned int nested_cnt = 0;
     for (;;) {
     cont:
         if (S_END(s)) {
```

Comment:
i seems that this is not reached. adding the nested_cnt check directly before label success was never reached (at least with my buggy json)